### PR TITLE
Build 3.0 site extension

### PR DESCRIPTION
**Pull request by BrennanConroy**
Start of https://github.com/aspnet/AspNetCore/issues/15381

### CODE REVIEW

**Review by bradygaster — APPROVED** [review]

File: src/SiteExtensions/Sdk/SiteExtension.targets [hunk 1]
```diff
@@ -50,8 +50,9 @@
         Project="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\HostingStartup.csproj"
         DepsFile="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\p\HostingStartup.deps.json"
         TrimmedDepsFile="$(_DepsOutputDirectory)%(HostingStartupPackageReference.Identity)\%(HostingStartupPackageReference.Identity).deps.json"
-        PackagePath="$(_BasePackagePath)\shared\Microsoft.AspNetCore.App\$(MicrosoftAspNetCoreAppPackageVersion)\"
+        PackagePath="$(_BasePackagePath)\shared\Microsoft.AspNetCore.App\$(AspNetCoreMajorMinorVersion).0\"
          />
+         <!-- Always use Major.Minor.0 so that if we have to produce a new SiteExtension during a patch build it will still work for non-patch runtimes. i.e. 3.0.0 dotnet will search for 3.0.0 and below and wouldn't find a 3.0.1 folder path -->
```
Comment (bradygaster) [inline src/SiteExtensions/Sdk/SiteExtension.targets]: 👍

**Comment by jkotalik** [comment]
@BrennanConroy are you able to get to this fairly soon? Personally I'd prefer to get this into the next patch available.

**Comment by BrennanConroy** [comment]
As far as I'm aware this is building correctly and effectively good to go.

We need to decide whether we are building it for 3.0 or skipping. And then figure out how to get the 3.0 bits into a 3.1 build.

**Comment by anurse** [comment]
> We need to decide whether we are building it for 3.0 or skipping. And then figure out how to get the 3.0 bits into a 3.1 build.

The Site Extension itself is "versionless" (i.e. users should always install the latest). I'd like us to build one out of `release/3.1` that includes bits for 2.1, 3.0 and 3.1 and ship that. Then we can revisit in 5.0 and hopefully not have to ship one there :).

**Comment by BrennanConroy** [comment]
> I'd like us to build one out of `release/3.1` that includes bits for 2.1, 3.0 and 3.1 and ship that

Yeah, we need to build and "release" the 3.0 one first. Then we can reference the 3.0 one in a 3.1 build.

**Comment by anurse** [comment]
Got it. Let's get this PR ready to take in March.

**Comment by BrennanConroy** [comment]
Closing in favor of https://github.com/dotnet/aspnetcore/pull/18568

**Pull request closed without merging**